File: web-crms/types/customer.ts
export type CustomerType = "regular" | "institutional_buyer";
export type CustomerStatus = "active" | "inactive" | "suspended";
export type MarketingChannel =
  | "email"
  | "sms"
  | "social_media"
  | "push_notification";
export type InteractionType = "sent" | "opened" | "clicked" | "converted";

export interface Customer {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  address?: string;
  profileImage?: string;
  customerType: CustomerType;
  status: CustomerStatus;
  notes?: string;
  isDeleted?: boolean;
  createdAt: string;
}

export interface MarketingHistory {
  id: string;
  title: string;
  description: string;
  channel: MarketingChannel;
  interactionType: InteractionType;
  sentAt: string;
}

export interface OrderItem {
  productName: string;
  quantity: number;
  price: number;
}

export interface Order {
  id: string;
  orderNumber: string;
  status: string;
  totalAmount: number;
  items: OrderItem[];
  createdAt: string;
}

export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  pageSize: number;
}

File: crms/CRM.Domain/Entities/Customer.cs
namespace CRM.Domain.Entities;

public enum CustomerStatus { Active, Inactive, Suspended }
public enum CustomerType { Regular, InstitutionalBuyer }

public class Customer
{
    public Guid Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Address { get; set; }
    public string? ProfileImage { get; set; }
    public string? Notes { get; set; }
    public string? AuthId { get; set; }
    public CustomerStatus Status { get; set; } = CustomerStatus.Active;
    public CustomerType CustomerType { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public bool IsDeleted { get; set; } = false;

    public ICollection<MarketingHistory> MarketingHistories { get; set; } = [];
}